src/Finder/SourceFilesFinder.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © 2017 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+declare(strict_types=1);
+
+namespace Infection\Finder;
+
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @author Lander Vanderstraeten <lander_vanderstraeten@hotmail.com>
+ */
+class SourceFilesFinder
+{
+    private $sourceDirectories;
+    private $excludeDirectories;
+
+    public function __construct(array $sourceDirectories, array $excludeDirectories)
+    {
+        $this->sourceDirectories = $sourceDirectories;
+        $this->excludeDirectories = $excludeDirectories;
+    }
+
+    public function getSourceFiles(string $filter = ''): Finder
+    {
+        $finder = new Finder();
+
+        array_walk($this->excludeDirectories, function ($excludePath) use ($finder) {
+            $finder->notPath($excludePath);
+        });
+
+        if (empty($filter)) {
```

Comment:
`'' !== $filter`, I recommend against using `empty()` whenever possible, it's a footgun function (`'0' _is_ empty)